pkg/services/datasources/service/store_test.go

```diff
@@ -56,13 +56,14 @@ func TestIntegrationDataAccess(t *testing.T) {
 			db := db.InitTestDB(t)
 			ss := SqlStore{db: db}
 			_, err := ss.AddDataSource(context.Background(), &datasources.AddDataSourceCommand{
-				OrgID:    10,
-				Name:     "laban",
-				Type:     datasources.DS_GRAPHITE,
-				Access:   datasources.DS_ACCESS_DIRECT,
-				URL:      "http://test",
-				Database: "site",
-				ReadOnly: true,
+				OrgID:      10,
+				Name:       "laban",
+				Type:       datasources.DS_GRAPHITE,
+				Access:     datasources.DS_ACCESS_DIRECT,
+				URL:        "http://test",
+				Database:   "site",
+				ReadOnly:   true,
+				APIVersion: "v0alpha1",
```

Comment:
Thoughts on where we should validate this?  When coming from an apiserver, the apiVersion is very explicit.  I'm not sure how we should validate the plumbing in our existing setup.  The most likely thing is that the existing UI should not change its payload (eg do note send apiversion), but if it seems valid, the currently configured one will be marked.

If the apiversion is set, but different than the plugin.json version.... it should... fail?  seems safest for now

Reply:
Since the UI is rendering a ConfigEditor for a specific version, I think it's legit for the client to set it in the payload since it's the version it's targetting.

Here in the backend, I am not sure if we need to do the check. In theory, the instance saving the datasource does not need to be the consumer. In any case, I think that having those two pieces with different API versions is highly improbable so I am okay if we want to add a validation check here.